spec/units/frame_spec.rb

```diff
@@ -0,0 +1,221 @@
+require 'frame'
+
+describe Frame do
+
+  subject(:frame) { Frame.new(1) }
+
+  let(:roll_2) { double('roll', pins: 2) }
+  let(:roll_3) { double('roll', pins: 3) }
+  let(:roll_8) { double('roll', pins: 8) }
+  let(:roll_10) { double('roll', pins: 10) }
```

Comment:
I'd put these in the tests themselves - tests don't need to be DRY, they mostly need to be readable.

https://exceptionalcode.wordpress.com/2010/08/23/do-repeat-yourself-in-unit-tests/

Reply:
What I'd like is to be able to read one test and understand everything that is happening in it, without having to read the full file.